src/backend/joanie/demo/management/commands/create_dev_demo.py

```diff
@@ -387,6 +389,37 @@ def handle(self, *args, **options):  # pylint: disable=too-many-locals
             )
         )
 
+        # Order for a PRODUCT_CREDENTIAL with a unsigned contract
+        order = self.create_product_purchased(
+            admin_user, organization, enums.PRODUCT_TYPE_CREDENTIAL
+        )
+        factories.ContractFactory(
+            order=order, definition=order.product.contract_definition, signed_on=None
+        )
+        self.stdout.write(
+            self.style.SUCCESS(
+                "Successfully create a order for a PRODUCT_CREDENTIAL with an unsigned contract"
+            )
+        )
+
+        # Order for a PRODUCT_CREDENTIAL with a signed contract
+        order = self.create_product_purchased(
+            admin_user, organization, enums.PRODUCT_TYPE_CREDENTIAL
+        )
+
+        factories.ContractFactory(
+            order=order,
+            definition=order.product.contract_definition,
+            definition_checksum=1234,
+            context={"foo": "bar"},
```

Comment:
For the purpose of the demo, I think you should generate a valid contract context.
https://github.com/openfun/joanie/blob/dead80989865da9398d005cf80e62c43d58a690e/src/backend/joanie/core/models/contracts.py#L73-L94